### DISCUSSION TITLE
How to know when a fetch is cached ?

### DISCUSSION THREAD

**Discussion by Djay77** [vercel/next.js#69893] [Help]
### Summary

Hello,

I need to know and log when a fetch() response came from cache.

How to check this ?

The Response status is still 200, cacheState is always an empty string and timingInfo is null !

<img width="567" alt="image" src="https://github.com/user-attachments/assets/9c566530-0be4-4e0a-bfc0-35bc20702080">

the same in dev or build + start

Next v.14.2.2

Thanks 
Djay

**Comment by mutasim77** [comment]
It's actually quite simple and it's a built-in feature in Next.js. You can enable it just by adding this logger to your Next.js config:
```js
const nextConfig = {
  logging: {
    fetches: {
      fullUrl: true
    }
  }
};
```

The result will look something like this:
```js
| GET http://example.com/api/data 200 in 123ms (cache: MISS)
| GET http://example.com/api/data 200 in 5ms (cache: HIT)
```
For requests that skip the cache, you might see:
```
| GET http://example.com/api/data 200 in 150ms (cache: SKIP)
    cache skipped reason: (cache: no-cache)
```

Hope this was helpful!

**Reply by Djay77** [reply]
Hello, thank you @mutasim77, I already have this config activated but I need to know this directly from the Response object (or something in my code to log the Request in an external service), not in my console :(

**Reply by jukkagronroos** [reply]
Hi, I encountered a similar problem, any solutions to this issue?

**Reply by ratz6** [reply]
any solution to this ?
